### ISSUE TITLE
[bug]: How are group folder permissions handled in context chat answers?

### ISSUE THREAD

**Issue by ga-it** [nextcloud/context_chat_backend#49]
**Describe the bug**
Context chat allows questions without a specified file or folder context. To the extent that this answer is produced from existing embeddings / vector database, what is the risk that this answer will include information from a context the user does not have access to?

This will not be impacted via referred source documents as if these include sources the user does not have access to, they will now be able to follow the link through to the document. But if contents from these documents are included in the answer, clearly this breaches security.

For example, if there is a group folder with salary information that is only accessible to finance staff, but embeddings are generated and then available for general query, this would breaches folder security.

Further, if this is correctly handled, are changes to a users permissions retroactively applied to previously generated embeddings?

I am uncertain if this is possible, but cannot see any documentation as to how this is handled.

**To Reproduce**
1. Go to AI Assistant, Context Chat
2. Leave "Selective context" unselected
3. Under "Input" ask a question regarding information the user does not have access to.

**Expected behavior**
Context chat answers should be based on a combination of the LLM used by Context Chat and only embeddings that are consistent with their Nextcloud permissions. Technically I am unsure how this would be accomplished, but conceptually imagine that a users permission matrix could be applied to embeddings to block answers from content they do not have access to.

**Setup Details (please complete the following information):**
 - Nextcloud Version: [28.0.5]
 - AppAPI Version: [2.6.0]
 - Context Chat PHP Version [8.2.18]
 - Context Chat Backend Version [2.1.0]
 - Nextcloud deployment method: [Nextcloud AIO]
 - Context Chat Backend deployment method: [ manual remote server]

**Additional context**

**Comment by marcelklehr** [comment]
Groupfolders' user access lists are respected when crawling for a user's data. What might not work yet, is adding a user to a groupfolder after the users files have been crawled.

**Comment by ga-it** [comment]
Thanks @marcelklehr - does that imply that the embeddings databases are separately created for each user?

I am not sure of the overhead of that design on a large corporate installation with TBs of data?

For that reason and changes in permissions, do you think the linked approach may work better - i.e. one embeddings database with permissions applied dynamically per user context?

**Comment by marcelklehr** [comment]
> does that imply that the embeddings databases are separately created for each user?

Yes, that's how it is at the moment. We are aware that this is not ideal.

**Comment by ga-it** [comment]
Is a change on the roadmap or should I file the above as a feature request?

**Comment by marcelklehr** [comment]
It's on the roadmap, yes :)

**Comment by ga-it** [comment]
Thanks @kyteinsky 

Do you have an ETA on that release?

**Comment by kyteinsky** [comment]
No, sorry, but it can take a few weeks.

**Comment by kyteinsky** [comment]
Fixed with the latest release.
